From 723ff2fa4708a211734f469e3cc2092beb6bbf02 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=A0=D1=83=D1=81=D0=BB=D0=B0=D0=BD=20=D0=98=D0=B6=D0=B1?=
 =?UTF-8?q?=D1=83=D0=BB=D0=B0=D1=82=D0=BE=D0=B2?= <lrn1986@gmail.com>
Date: Mon, 13 Apr 2015 15:31:56 +0000
Subject: [PATCH 11/20] [W32] Change OpenGL checks to work on W32, add
 libopengl32

AGain, use OS(WINDOWS) for things that are W32 but not necessarily MSVC.
Don't try to check for dlopen() on W32 (it might be available, but
native code is already here, so use it).

Correctly cast the result of GetProcAddress(), because C++.

Link W32 version to -lOpenGL32 instead of -lGL, otherwise libtool complains:
*** Warning: linker path does not have real file for library -lGL.
---
 Source/WebCore/platform/graphics/OpenGLShims.cpp | 6 +++---
 Source/autotools/FindDependencies.m4             | 7 ++++++-
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/Source/WebCore/platform/graphics/OpenGLShims.cpp b/Source/WebCore/platform/graphics/OpenGLShims.cpp
index 916a908..d77a8b0 100644
--- a/Source/WebCore/platform/graphics/OpenGLShims.cpp
+++ b/Source/WebCore/platform/graphics/OpenGLShims.cpp
@@ -22,7 +22,7 @@
 #define DISABLE_SHIMS
 #include "OpenGLShims.h"
 
-#if !PLATFORM(WIN)
+#if !OS(WINDOWS)
 #include <dlfcn.h>
 #endif
 
@@ -37,10 +37,10 @@ OpenGLFunctionTable* openGLFunctionTable()
     return &table;
 }
 
-#if PLATFORM(WIN)
+#if OS(WINDOWS)
 static void* getProcAddress(const char* procName)
 {
-    return GetProcAddress(GetModuleHandleA("libGLESv2"), procName);
+    return reinterpret_cast<void*>(GetProcAddress(GetModuleHandleA("libGLESv2"), procName));
 }
 #else
 typedef void* (*glGetProcAddressType) (const char* procName);
diff --git a/Source/autotools/FindDependencies.m4 b/Source/autotools/FindDependencies.m4
index 9a2c496..3ce3f76 100644
--- a/Source/autotools/FindDependencies.m4
+++ b/Source/autotools/FindDependencies.m4
@@ -453,6 +453,9 @@ if test "$found_opengl" = "yes"; then
     if test "$enable_gles2" = "yes"; then
         acceleration_description="$acceleration_description (gles2"
         OPENGL_LIBS="-lGLESv2"
+    elif test "$os_win32" = "yes"; then
+        acceleration_description="$acceleration_description (gl"
+        OPENGL_LIBS="-lopengl32"
     else
         acceleration_description="$acceleration_description (gl"
         OPENGL_LIBS="-lGL"
@@ -467,7 +470,9 @@ if test "$found_opengl" = "yes"; then
 
     # Check whether dlopen() is in the core libc like on FreeBSD, or in a separate
     # libdl like on GNU/Linux (in which case we want to link to libdl).
-    AC_CHECK_FUNC([dlopen], [], [AC_CHECK_LIB([dl], [dlopen], [OPENGL_LIBS="$OPENGL_LIBS -ldl"])])
+    if test "$os_win32" = "no"; then
+        AC_CHECK_FUNC([dlopen], [], [AC_CHECK_LIB([dl], [dlopen], [OPENGL_LIBS="$OPENGL_LIBS -ldl"])])
+    fi
 
     acceleration_description="$acceleration_description)"
 fi
-- 
1.8.5.3

